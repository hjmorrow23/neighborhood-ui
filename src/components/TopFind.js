import React from 'react';
import img from "../assets/img/house1.jpg";


const TopFind = props => {
    return (
        
            <a href="" className="top-find">
                <img className="top-find-image" src={img} />
                <p className="top-find-price">{props.listing.price}</p>
                <p className="top-find-address">{props.listing.address}</p>
            </a>
        
    );
};

export default TopFind;