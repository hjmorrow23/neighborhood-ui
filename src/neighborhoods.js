import img1 from "./assets/img/house1.jpg";
import img2 from "./assets/img/house2.jpg";
import img3 from "./assets/img/house3.jpg";
import img4 from "./assets/img/house4.jpg";
import img5 from "./assets/img/house5.jpg";
import img6 from "./assets/img/house6.jpg";

const neighborhoods = [
    {
        name: "Beverly",
        city: "Chicago",
        state: "IL",
        agent: "John Doe",
        overview: "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
        rating: "5.0",
        starts: 5,
        topListings: [
            {
                address: "4 Privet Drive",
                price: "$375,000",
                image: img1
            },
            {
                address: "1060 West Addison Street",
                price: "$290,000",
                image: img2
            },
            {
                address: "1323 Mockingbird Lane",
                price: "310,000",
                image: img3
            },
        ]
    },
    {
        name: "Oakland",
        city: "Chicago",
        state: "IL",
        agent: "Jane Doe",
        overview: "Bordering the coast of Lake Michigan, Oakland is a quiet neighborhood on Chicago's south side. With scenic views of the glittering lake and Chicago's towering skyline, this promising neighborhood is sure to continue growing in the coming years.",
        rating: "5.0",
        starts: 5,
        topListings: [
            {
                address: "221B Baker Street",
                price: "$375,000",
                image: img4
            },
            {
                address: "425 Grove Street",
                price: "$290,000",
                image: img5
            },
            {
                address: "124 Conch Street",
                price: "310,000",
                image: img6
            },
        ]
    },
];

export default neighborhoods;