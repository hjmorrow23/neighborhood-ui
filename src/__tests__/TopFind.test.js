import React, { Component } from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TopFind from "../components/TopFind";
import toJSON from 'enzyme-to-json';
import neighborhoods from '../neighborhoods';

configure({ adapter: new Adapter() });

const listing = neighborhoods[0];

describe('<Dashboard />', () => {
    it('renders', () => {
        mount(
            <TopFind listing={listing} />
        );
    });

    it('matches the snapshot', () => {
        const wrapper = mount(
            <TopFind listing={listing} />
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('updates via props', () => {
        const wrapper = mount(
            <TopFind listing={listing} count={50} />
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.setProps({ count: 10 });
        expect(toJSON(wrapper)).toMatchSnapshot();
      });
});
