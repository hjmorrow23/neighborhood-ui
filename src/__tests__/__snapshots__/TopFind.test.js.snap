// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> matches the snapshot 1`] = `
<TopFind
  listing={
    Object {
      "agent": "John Doe",
      "city": "Chicago",
      "name": "Beverly",
      "overview": "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
      "rating": "5.0",
      "starts": 5,
      "state": "IL",
      "topListings": Array [
        Object {
          "address": "4 Privet Drive",
          "image": "house1.jpg",
          "price": "$375,000",
        },
        Object {
          "address": "1060 West Addison Street",
          "image": "house2.jpg",
          "price": "$290,000",
        },
        Object {
          "address": "1323 Mockingbird Lane",
          "image": "house3.jpg",
          "price": "310,000",
        },
      ],
    }
  }
>
  <a
    className="top-find"
    href=""
  >
    <img
      className="top-find-image"
      src="house1.jpg"
    />
    <p
      className="top-find-price"
    />
    <p
      className="top-find-address"
    />
  </a>
</TopFind>
`;

exports[`<Dashboard /> updates via props 1`] = `
<TopFind
  count={50}
  listing={
    Object {
      "agent": "John Doe",
      "city": "Chicago",
      "name": "Beverly",
      "overview": "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
      "rating": "5.0",
      "starts": 5,
      "state": "IL",
      "topListings": Array [
        Object {
          "address": "4 Privet Drive",
          "image": "house1.jpg",
          "price": "$375,000",
        },
        Object {
          "address": "1060 West Addison Street",
          "image": "house2.jpg",
          "price": "$290,000",
        },
        Object {
          "address": "1323 Mockingbird Lane",
          "image": "house3.jpg",
          "price": "310,000",
        },
      ],
    }
  }
>
  <a
    className="top-find"
    href=""
  >
    <img
      className="top-find-image"
      src="house1.jpg"
    />
    <p
      className="top-find-price"
    />
    <p
      className="top-find-address"
    />
  </a>
</TopFind>
`;

exports[`<Dashboard /> updates via props 2`] = `
<TopFind
  count={10}
  listing={
    Object {
      "agent": "John Doe",
      "city": "Chicago",
      "name": "Beverly",
      "overview": "Beverly is famous for its architecture, extensive green spaces, and historic charm. This South Side neighborhood is also known for its Irish-style pubs along Western Avenue, as well as its unique shopping district that stretches down 103rd Street. Beverly offers an eclectic environment like a big city, yet provides a distinct small-town vibe. The neighborhood has a variety of homes with plenty of backyard space for outdoor recreation.",
      "rating": "5.0",
      "starts": 5,
      "state": "IL",
      "topListings": Array [
        Object {
          "address": "4 Privet Drive",
          "image": "house1.jpg",
          "price": "$375,000",
        },
        Object {
          "address": "1060 West Addison Street",
          "image": "house2.jpg",
          "price": "$290,000",
        },
        Object {
          "address": "1323 Mockingbird Lane",
          "image": "house3.jpg",
          "price": "310,000",
        },
      ],
    }
  }
>
  <a
    className="top-find"
    href=""
  >
    <img
      className="top-find-image"
      src="house1.jpg"
    />
    <p
      className="top-find-price"
    />
    <p
      className="top-find-address"
    />
  </a>
</TopFind>
`;
